python/ray/serve/_private/client.py

```diff
@@ -287,9 +292,25 @@ def deploy_group(
             self.delete_deployments(deployment_names_to_delete, blocking=_blocking)
 
     @_ensure_connected
-    def deploy_app(self, config: ServeApplicationSchema) -> None:
+    def deploy_app(
+        self, config: ServeApplicationSchema, _blocking: bool = False
+    ) -> None:
         ray.get(self._controller.deploy_app.remote(config))
 
+        if _blocking:
+            timeout_s = 60
+
+            start = time.time()
+            while time.time() - start < timeout_s:
```

Comment:
The `while` loop and `if` conditionals can be combined:

```suggestion
            while time.time() - start < timeout_s and self.get_serve_status().status != ApplicationStatus.RUNNING:
```

If you make this change, make sure to remove the `if`-`break` clause.